### PR TITLE
Barkha_Assignment_22_25 mins

### PR DESCRIPTION
**Pull request by BarkhaPJun20**
Program 1: 
Verify given number is palindrome or not. 
Input: number = 45654
Output : Given number 45654 is palindrome.

Input: number = 56789
Output : Given number 56789 is not palindrome.

Program 2:
Verify given String is palindrome or not. 
Input: radar
Output : Given string radar is palindrome.

Input: technocredits
Output: Given string technocredits is not palindrome
image

image

### CODE REVIEW

File: src/barkhaPJun20/PalindromeNumber.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+			num1=num1/10;
+		}
+		if (reverseNum==originalNum)
+			System.out.println("Number ("+originalNum+") is a Palindrome Number.");
+		else
+			System.out.println("Number ("+originalNum+") is Not a Palindrome Number.");
+	}
+	public static void main(String[] args) {
+		PalindromeNumber palindromeNumber=new PalindromeNumber();
+		int num=45654;
```
Comment (SwapnaWJun20) [inline src/barkhaPJun20/PalindromeNumber.java]: You can pass the number directly while calling the method.
palindromeNumber.palindromeNumber(45654);

File: src/barkhaPJun20/PalindromeString.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+			temp=temp+name1.charAt(index);   //String+anything(here character) will be String.
+		}
+		if (temp.equalsIgnoreCase(originalName))
+			System.out.println("Name ("+originalName+") is a Palindrome Name.");
+		else
+			System.out.println("Name ("+originalName+") is Not a Palindrome Name.");
+	}
+	public static void main(String[] args) {
+		PalindromeString palindromeString=new PalindromeString();
+		String name="radar";
```
Comment (SwapnaWJun20) [inline src/barkhaPJun20/PalindromeString.java]: You can pass the string directly while calling the method.
palindromeString.palindromeString("radar");

File: src/barkhaPJun20/PalindromeNumber.java (hunk 1 above)
Comment (BarkhaPJun20) [reply src/barkhaPJun20/PalindromeNumber.java]: Changes Done

File: src/barkhaPJun20/PalindromeString.java (hunk 2 above)
Comment (BarkhaPJun20) [reply src/barkhaPJun20/PalindromeString.java]: Changes Done

**Review by SwapnaWJun20 — APPROVED** [review]

**Comment by SwapnaWJun20** [comment]
Code looks good to me

File: src/barkhaPJun20/PalindromeNumber.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+public class PalindromeNumber {
+	
+	void palindromeNumber(int originalNum) {
+		int num1=originalNum;                //num1=45654
+		int lastNum;
+		int reverseNum=0;
+		while(num1>0) {
+			lastNum=num1%10;                //lastNum=4
+			reverseNum=(reverseNum*10)+lastNum;
```
Comment (KrishnaKTechnocredits) [inline src/barkhaPJun20/PalindromeNumber.java:11]: lastNum variable can be removed.

reverseNum=(reverseNum*10)+ (num1%10)

File: src/barkhaPJun20/PalindromeString.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package barkhaPJun20;
+
+public class PalindromeString {
+	
+	void palindromeString (String originalName) {
+		String name1=originalName;
```
Comment (KrishnaKTechnocredits) [inline src/barkhaPJun20/PalindromeString.java:6]: You have not changed name1 value anywhere in the program. no need to assign originalName to name1. if you are changing originalName value then you have to store to it.

**Review by KrishnaKTechnocredits — APPROVED** [review]
Code looks good to me. Please take a look at my comment.

**Pull request merged**